support-frontend/app/models/identity/responses/IdentityErrorResponse.scala

```diff
@@ -0,0 +1,47 @@
+package models.identity.responses
+
+import models.identity.responses.IdentityErrorResponse.IdentityError
+import play.api.libs.json.{Json, Reads}
+
+// Models and error returned by Identity
+//
+// Example response:
+// =================
+//
+// {
+//    "status": "error",
+//    "errors": [
+//        {
+//            "message": "Registration Error",
+//            "description": "Please sign up using an email address from a different provider"
+//        }
+//    ]
+//}
+
+case class IdentityErrorResponse(
+  status: String,
+  errors: List[IdentityError]
+)
+
+object IdentityErrorResponse {
+  implicit val reads: Reads[IdentityErrorResponse] = Json.reads[IdentityErrorResponse]
+
+  case class IdentityError(
+    message: String,
+    description: String
+  )
+
+  object IdentityError {
+    object InvalidEmailAddress {
+      val message = "Registration Error"
+      val description = "Please sign up using an email address from a different provider"
+      val errorReasonCode = "invalid_email_address"
+    }
+
+    def isDisallowedEmailError(identityError: IdentityError): Boolean =
+      identityError.message == InvalidEmailAddress.message &&
+      identityError.description == InvalidEmailAddress.description
```

Comment:
@rupertbates you only need to look at the "message"="Registration Error", the description is flexible and we designed it to be returned to the client directly